src/pages/home/sidebar/SidebarLinksData.js

```diff
@@ -66,17 +66,21 @@ const defaultProps = {
 function SidebarLinksData({isFocused, allReportActions, betas, chatReports, currentReportID, insets, isLoadingReportData, isSmallScreenWidth, onLinkClick, policies, priorityMode}) {
     const {translate} = useLocalize();
 
-    const reportIDsRef = useRef([]);
+    const reportIDsRef = useRef(null);
+    const isLoading = SessionUtils.didUserLogInDuringSession() && isLoadingReportData;
     const optionListItems = useMemo(() => {
         const reportIDs = SidebarUtils.getOrderedReportIDs(currentReportID, chatReports, betas, policies, priorityMode, allReportActions);
         if (deepEqual(reportIDsRef.current, reportIDs)) {
             return reportIDsRef.current;
         }
-        reportIDsRef.current = reportIDs;
-        return reportIDs;
-    }, [allReportActions, betas, chatReports, currentReportID, policies, priorityMode]);
-
-    const isLoading = SessionUtils.didUserLogInDuringSession() && isLoadingReportData;
+        if (isLoading && !reportIDsRef.current && !currentReportID) {
+            reportIDsRef.current = reportIDs;
+        }
+        if (!isLoading) {
+            reportIDsRef.current = reportIDs;
+        }
+        return reportIDsRef.current || [];
+    }, [allReportActions, betas, chatReports, currentReportID, policies, priorityMode, isLoading]);
```

Comment:
@s-alves10 can you add some comments trying to say why we using this ref and notes on each of the cases you have there?

It is just not very easy to understand why we have this complexity

Reply:
Comments added